src/mito2/src/cache/cache_size.rs

```diff
@@ -137,6 +137,6 @@ mod tests {
     fn test_parquet_meta_size() {
         let metadata = parquet_meta();
 
-        assert_eq!(964, parquet_meta_size(&metadata));
+        assert_eq!(956, parquet_meta_size(&metadata));
```

Comment:
How's updating the rust affects this test result?

Reply:
> How's updating the rust affects this test result?

Good question. Because rustc now use niche encoding for enums like [`parquet::schema::types::Type`](https://docs.rs/parquet/53.2.0/parquet/schema/types/enum.Type.html) so we can spare 1 byte for variant tagging.

The related PR is https://github.com/rust-lang/rust/pull/130508

Also for the difference in layout:
Previously:
```
layout_of(Type) = Layout {
           size: Size(80 bytes),
           align: AbiAndPrefAlign {
               abi: Align(8 bytes),
               pref: Align(8 bytes),
           },
           abi: Aggregate {
               sized: true,
           },
           fields: Arbitrary {
               offsets: [
                   Size(0 bytes),
               ],
               memory_index: [
                   0,
               ],
           },
           largest_niche: Some(
               Niche {
                   offset: Size(0 bytes),
                   value: Int(
                       I8,
                       false,
                   ),
                   valid_range: 0..=1,
               },
           ),
           variants: Multiple {
               tag: Initialized {
                   value: Int(
                       I8,
                       false,
                   ),
                   valid_range: 0..=1,
               },
               tag_encoding: Direct,
               tag_field: 0,
               variants: [
                   Layout {
                       size: Size(64 bytes),
                       align: AbiAndPrefAlign {
                           abi: Align(8 bytes),
                           pref: Align(8 bytes),
                       },
                       abi: Aggregate {
                           sized: true,
                       },
                       fields: Arbitrary {
                           offsets: [
                               Size(16 bytes),
                               Size(1 bytes),
                               Size(4 bytes),
                               Size(8 bytes),
                               Size(12 bytes),
                           ],
                           memory_index: [
                               4,
                               0,
                               1,
                               2,
                               3,
                           ],
                       },
                       largest_niche: Some(
                           Niche {
                               offset: Size(16 bytes),
                               value: Int(
                                   I64,
                                   false,
                               ),
                               valid_range: 0..=9223372036854775807,
                           },
                       ),
                       variants: Single {
                           index: 0,
                       },
                       max_repr_align: None,
                       unadjusted_abi_align: Align(8 bytes),
                   },
                   Layout {
                       size: Size(80 bytes),
                       align: AbiAndPrefAlign {
                           abi: Align(8 bytes),
                           pref: Align(8 bytes),
                       },
                       abi: Aggregate {
                           sized: true,
                       },
                       fields: Arbitrary {
                           offsets: [
                               Size(8 bytes),
                               Size(56 bytes),
                           ],
                           memory_index: [
                               0,
                               1,
                           ],
                       },
                       largest_niche: Some(
                           Niche {
                               offset: Size(8 bytes),
                               value: Int(
                                   I64,
                                   false,
                               ),
                               valid_range: 0..=9223372036854775807,
                           },
                       ),
                       variants: Single {
                           index: 1,
                       },
                       max_repr_align: None,
                       unadjusted_abi_align: Align(8 bytes),
                   },
               ],
           },
           max_repr_align: None,
           unadjusted_abi_align: Align(8 bytes),
       }
```

With niche encoding:
```
layout_of(Type) = Layout {
           size: Size(72 bytes),
           align: AbiAndPrefAlign {
               abi: Align(8 bytes),
               pref: Align(8 bytes),
           },
           abi: Aggregate {
               sized: true,
           },
           fields: Arbitrary {
               offsets: [
                   Size(0 bytes),
               ],
               memory_index: [
                   0,
               ],
           },
           largest_niche: Some(
               Niche {
                   offset: Size(0 bytes),
                   value: Int(
                       I64,
                       false,
                   ),
                   valid_range: 0..=9223372036854775808,
               },
           ),
           variants: Multiple {
               tag: Initialized {
                   value: Int(
                       I64,
                       false,
                   ),
                   valid_range: 0..=9223372036854775808,
               },
               tag_encoding: Niche {
                   untagged_variant: 1,
                   niche_variants: 0..=0,
                   niche_start: 9223372036854775808,
               },
               tag_field: 0,
               variants: [
                   Layout {
                       size: Size(72 bytes),
                       align: AbiAndPrefAlign {
                           abi: Align(8 bytes),
                           pref: Align(8 bytes),
                       },
                       abi: Aggregate {
                           sized: true,
                       },
                       fields: Arbitrary {
                           offsets: [
                               Size(8 bytes),
                               Size(68 bytes),
                               Size(56 bytes),
                               Size(60 bytes),
                               Size(64 bytes),
                           ],
                           memory_index: [
                               0,
                               4,
                               1,
                               2,
                               3,
                           ],
                       },
                       largest_niche: None,
                       variants: Single {
                           index: 0,
                       },
                       max_repr_align: None,
                       unadjusted_abi_align: Align(8 bytes),
                   },
                   Layout {
                       size: Size(72 bytes),
                       align: AbiAndPrefAlign {
                           abi: Align(8 bytes),
                           pref: Align(8 bytes),
                       },
                       abi: Aggregate {
                           sized: true,
                       },
                       fields: Arbitrary {
                           offsets: [
                               Size(0 bytes),
                               Size(48 bytes),
                           ],
                           memory_index: [
                               0,
                               1,
                           ],
                       },
                       largest_niche: Some(
                           Niche {
                               offset: Size(0 bytes),
                               value: Int(
                                   I64,
                                   false,
                               ),
                               valid_range: 0..=9223372036854775807,
                           },
                       ),
                       variants: Single {
                           index: 1,
                       },
                       max_repr_align: None,
                       unadjusted_abi_align: Align(8 bytes),
                   },
               ],
           },
           max_repr_align: None,
           unadjusted_abi_align: Align(8 bytes),
       }
```